packages/blocks/plus_minus/src/index.js

```diff
@@ -0,0 +1,1048 @@
+    'procedure_context_menu', procedureContextMenu);
+
+const procedureDefMutator = {
+  /**
+   * Create XML to represent the argument inputs.
+   * @param {boolean=} opt_isForCaller If true include the procedure name and
+   *     argument IDs. Used by Blockly.Procedures.mutateCallers for
+   *     reconnection.
+   * @return {!Element} XML storage element.
+   * @this Blockly.Block
+   */
+  mutationToDom: function(opt_isForCaller) {
+    const container = Blockly.utils.xml.createElement('mutation');
+    if (opt_isForCaller) {
+      container.setAttribute('name', this.getFieldValue('NAME'));
+    }
+    for (let i = 0; i < this.argumentVarModels_.length; i++) {
+      const argument = Blockly.utils.xml.createElement('arg');
+      const argModel = this.argumentVarModels_[i];
+      argument.setAttribute('name', argModel.name);
+      argument.setAttribute('varid', argModel.getId());
+      if (opt_isForCaller) {
+        argument.setAttribute('paramid', this.argIds_[i]);
+      }
+      container.appendChild(argument);
+    }
+
+    // Not used by this block, but necessary if switching back and forth
+    // between this mutator UI and the default UI.
+    if (!this.hasStatements_) {
+      container.setAttribute('statements', 'false');
+    }
+
+    return container;
+  },
+
+  /**
+   * Parse XML to restore the argument inputs.
+   * @param {!Element} xmlElement XML storage element.
+   * @this Blockly.Block
+   */
+  domToMutation: function(xmlElement) {
+    // We have to handle this so that the user doesn't add blocks to the stack,
+    // in which case it would be impossible to return to the old mutators.
+    this.hasStatements_ = xmlElement.getAttribute('statements') !== 'false';
+    if (!this.hasStatements_) {
+      this.removeInput('STACK');
+    }
+
+    const names = [];
+    const ids = [];
+    for (let i = 0, childNode; (childNode = xmlElement.childNodes[i]); i++) {
+      if (childNode.nodeName.toLowerCase() != 'arg') {
+        continue;
+      }
+      names.push(childNode.getAttribute('name'));
+      ids.push(childNode.getAttribute('varid') ||
+          childNode.getAttribute('varId'));
+    }
+    this.updateShape_(names, ids);
+  },
+
+  /**
+   * Adds arguments to the block until it matches the targets.
+   * @param {!Array<string>} names An array of argument names to display.
+   * @param {!Array<string>} varIds An array of variable IDs associated with
+   *     those names.
+   * @this Blockly.Block
+   * @private
+   */
+  updateShape_: function(names, varIds) {
+    // In this case it is easiest to just reset and build from scratch.
+
+    // We need to remove args in reverse order so that it doesn't mess up
+    // as removeArg_ modifies our arrays.
+    for (let i = this.argIds_.length - 1; i >= 0; i--) {
+      this.removeArg_(this.argIds_[i]);
+    }
+
+    this.arguments_ = [];
+    this.varIds_ = [];
+    this.argumentVarModels_ = [];
+    this.argIds_ = [];
+
+    const length = varIds.length;
```

Comment:
Check that `varids.length` and `names.length` are the same, and error if not, unless guaranteed elsewhere.